File: Cafebook/CafebookModel/Model/ModelWeb/KhachHang/DangNhapDto.cs
namespace CafebookModel.Model.ModelWeb.KhachHang
{
    // DTO gửi lên từ View
    public class DangNhapRequestDto
    {
        public string TenDangNhap { get; set; } = string.Empty;
        public string MatKhau { get; set; } = string.Empty;
    }

    // DTO trả về từ API
    public class DangNhapResponseDto
    {
        public bool Success { get; set; }
        public string? Message { get; set; }
        public DangNhapKhachHangDto? KhachHangData { get; set; }
        public string? Token { get; set; }
    }

    // DTO chứa dữ liệu User an toàn
    public class DangNhapKhachHangDto
    {
        public int IdKhachHang { get; set; }
        public string HoTen { get; set; } = string.Empty;
        public string? Email { get; set; }
        public string? SoDienThoai { get; set; }
        public string? TenDangNhap { get; set; }
        public string? AnhDaiDienUrl { get; set; }
    }
}

File: Cafebook/ADRCafebookApi/app/src/main/java/com/example/cafebook/models/DangNhapDto.java
package com.example.cafebook.models;

import com.google.gson.annotations.SerializedName;

public class DangNhapDto {

    public static class Request {
        @SerializedName("TenDangNhap") public String tenDangNhap;
        @SerializedName("MatKhau") public String matKhau;

        public Request(String tenDangNhap, String matKhau) {
            this.tenDangNhap = tenDangNhap;
            this.matKhau = matKhau;
        }
    }

    public static class Response {
        @SerializedName("success") public boolean success;
        @SerializedName("message") public String message;
        @SerializedName("khachHangData") public KhachHangData khachHangData;
        @SerializedName("token") public String token;
    }

    public static class KhachHangData {
        @SerializedName("idKhachHang") public int idKhachHang;
        @SerializedName("hoTen") public String hoTen;
        @SerializedName("email") public String email;
        @SerializedName("soDienThoai") public String soDienThoai;
        @SerializedName("tenDangNhap") public String tenDangNhap;
        @SerializedName("anhDaiDienUrl") public String anhDaiDienUrl;
    }
}
